ui-tests/src/test/java/org/openmrs/reference/page/ManageLocationsPage.java

```diff
@@ -15,10 +15,24 @@
 
 public class ManageLocationsPage extends Page {
 
-    public static final By CONFIRM_PURGE_BUTTON = By.cssSelector("#adminui-purge-location-dialog > div.dialog-content > form > button.confirm.right");
+    protected static final String PAGE_URL = "/adminui/metadata/locations/manageLocations.page";
+    private static final String PARENT_LOCATION = "Pharmacy";
+    private static final By EDIT_LOCATION_LINK = By.cssSelector("i.icon-pencil.edit-action");
+    public static final By CONFIRM_PURGE_BUTTON = By.cssSelector("#adminui-purge-location-dialog button.confirm.right");
     public static final By CONFIRM_RETIRE_BUTTON = By.cssSelector("#adminui-retire-location-dialog > div.dialog-content > form > button.confirm.right");
-    protected static final String PAGE_URL = "adminui/metadata/locations/manageLocations.page";
+    private static final By LOCATION_TAG_BUTTON = By.cssSelector("#locationForm fieldset table tbody tr:first-child td:first-child input[type=\"checkbox\"]");
+    private static final By ADD_NEW_LOCATION_BUTTON = By.xpath("//*[@id='content']/a");
+    private static final By LOCATION_NAME_FIELD = By.id("name-field");  
     private static final By ADD_EDIT_LOCATION = By.cssSelector("#content a[href='/openmrs/" + AddEditLocationPage.PAGE_URL + "']");
+    private static final By LOCATION_DESCRIPTION_FIELD = By.cssSelector("#description-field");
+    private static final By LOCATION_ADDRESS_FIELD_1 = By.cssSelector("#address1-field");
+    private static final By LOCATION_ADDRESS_FIELD_2 = By.cssSelector("#address2-field");
+    private static final By CITY_FIELD = By.cssSelector("#cityVillage-field");
+    private static final By STATE_FIELD = By.cssSelector("#stateProvince-field");
+    private static final By COUNTRY_FIELD = By.cssSelector("#country-field");
+    private static final By POSTAL_CODE_FIELD = By.cssSelector("#postalCode-field");
+    private static final By PARENT_LOCATION_FIELD = By.cssSelector("#fr7903-field");
```

Comment:
@kdaud can you point me to the element with this id `fr7903-field`?

Reply:
>  can you point me to the element with this id fr7903-field?

![cssSelector](https://user-images.githubusercontent.com/38831673/133107928-76f96be3-807a-4d1f-b402-d28fa98c7b01.png)

@dkayiwa The page element Parent Location  has an `id = fr7903-field`. But I opted using its `cssSelector()`

Reply:
A github link to the code would serve better than an image.

Reply:
Located at  https://github.com/openmrs/openmrs-distro-referenceapplication/blob/da069437d917b292e7bae87fe4d5f11c4dfee107/ui-tests/src/test/java/org/openmrs/reference/page/ManageLocationsPage.java#L34

Reply:
I mean a link to the actual element. Not your code that references the element.

Reply:
The element(Parent Location) is within the form located at https://qa-refapp.openmrs.org/openmrs/adminui/metadata/locations/location.page

Reply:
Is it a runtime automatically generated id?

Reply:
Its a runtime generated id and isn't a good idea to use for the element. Am looking for a static element reference that can hardly break.